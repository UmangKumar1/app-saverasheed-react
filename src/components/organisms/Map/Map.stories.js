import React from "react"
import { storiesOf } from "@storybook/react"
import { action } from "@storybook/addon-actions"
import Map from "."

const token = process.env.STORYBOOK_MAPBOX_API_TOKEN

storiesOf("Map", module).add("default", () => (
  <Map
    {...{
      entityClick: action(""),
      selectEntity: action(""),
      deselectEntity: action(""),
      selectedEntities: [],
      viewport: {
        latitude: 37.785164,
        longitude: -122.41669,
        zoom: 17,
        bearing: 0,
        pitch: 60,
        width: 500,
        height: 500,
      },
      changeViewport: action(""),
      popupInfo: {},
      closePopup: action(""),
      userLocation: { latitude: 1, longitude: 1 },
      token,
      pois: [],
    }}
  />
))
