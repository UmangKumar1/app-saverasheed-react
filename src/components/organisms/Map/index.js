export { default, default as Map } from './Map'
