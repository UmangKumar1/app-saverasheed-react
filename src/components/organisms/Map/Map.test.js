import React from 'react'
import { shallow } from 'enzyme'
import Map from '.'

const wrap = (props = {}) => shallow(<Map {...props} />)

describe('Map', () => {
  it('renders children when passed in', () => {
    const wrapper = wrap({
      entityClick: jest.fn(),
      selectEntity: jest.fn(),
      deselectEntity: jest.fn(),
      selectedEntities: [],
      viewport: {},
      changeViewport: jest.fn(),
      popupInfo: {},
      fetchQuest: jest.fn(),
      closePopup: jest.fn(),
      userLocation: { latitude: 0, longitude: 0 },
      pois: [],
    })
    expect(wrapper).toMatchSnapshot()
  })
})
