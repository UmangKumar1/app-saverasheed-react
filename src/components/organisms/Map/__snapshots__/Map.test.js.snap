// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map renders children when passed in 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapPage
    changeViewport={[MockFunction]}
    closePopup={[MockFunction]}
    deselectEntity={[MockFunction]}
    entityClick={[MockFunction]}
    fetchQuest={[MockFunction]}
    pois={Array []}
    popupInfo={Object {}}
    selectEntity={[MockFunction]}
    selectedEntities={Array []}
    userLocation={
      Object {
        "latitude": 0,
        "longitude": 0,
      }
    }
    viewport={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": [Function],
      "disableHeight": false,
      "disableWidth": false,
      "onResize": [Function],
      "style": Object {},
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": [Function],
        "disableHeight": false,
        "disableWidth": false,
        "onResize": [Function],
        "style": Object {},
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
